services/web-server/src/graphql/Hooks.graphql

```diff
@@ -89,11 +95,22 @@ input HookInput {
   triggerSchema: JSON
 }
 
+type HookLastFire {
+  hookGroupId: ID!,
+  hookId: ID!,
+  firedBy: HookFiredBy!,
+  taskId: String!,
+  taskCreateTime: DateTime!,
+  result:HookFireResult!,
+  error: String!,
+}
+
 extend type Query {
   hookGroups(filter: JSON): [HookGroup]
   hooks(hookGroupId: ID!, filter: JSON): [Hook]
   hook(hookGroupId: ID!, hookId: ID!): Hook
   hookStatus(hookGroupId: ID!, hookId: ID!): HookStatus
+  hookLastFires(hookGroupId: ID!, hookId: ID!): [HookLastFire]
```

Comment:
When the response is an array, we should allow a user to filter on the result. Let's add an additional optional parameter `filter: JSON`. See the `diff` for how to fully integrate the filter feature.

Reply:
So I think of adding a searchbar for finding apt lastfires by success, time etc.

Reply:
The graphQL endpoint should still have a filter parameter. It doesn't mean we will use it but as a convention, for all response returning an array, we should expose a filter parameter.